subgraphs/_reference_/src/sdk/protocols/bridge/README.md

```diff
@@ -0,0 +1,375 @@
+import { Pool } from "./sdk/protocols/bridge/pool";
+import { BridgeConfig } from "./sdk/protocols/bridge/config";
+import { _ERC20 } from "wherever You have an ABI for it";
+import { Versions } from "./versions";
+
+// Implement TokenPricer to pass it to the SDK constructor
+class Pricer implements TokenPricer {
+  getTokenPrice(token: Token): BigDecimal {
+    const price = getUsdPricePerToken(Address.fromBytes(token.id));
+    return price.usdPrice;
+  }
+
+  getAmountPrice(token: Token, amount: BigInt): BigDecimal {
+    const _amount = bigIntToBigDecimal(amount, token.decimals);
+    return getUsdPrice(Address.fromBytes(token.id), _amount);
+  }
+}
+
+// Implement TokenInitializer
+class TokenInit implements TokenInitializer {
+  getTokenParams(address: Address): TokenParams {
+    const erc20 = _ERC20.bind(address);
+    const name = erc20.name();
+    const symbol = erc20.symbol();
+    const decimals = erc20.decimals().toI32();
+    return {
+      name,
+      symbol,
+      decimals,
+    };
+  }
+}
+
+const conf = new BridgeConfig(
+  "0x2796317b0fF8538F253012862c06787Adfb8cEb6",
+  "Synapse",
+  "synapse",
+  "WHITELIST",
+  Versions
+);
+
+export function handleTransferOut(event: TransferOut): void {
+  const sdk = new SDK(conf, new Pricer(), new TokenInit(), event);
+
+  const poolID = event.address;
+  const pool = sdk.Pools.loadPool(
+    poolID,
+    onCreatePool,
+    BridgePoolType.LOCK_RELEASE
+  );
+  const crossToken = sdk.Tokens.initCrosschainToken(
+    event.params.chainId,
+    event.params.token,
+    CrosschainTokenType.WRAPPED,
+    event.params.token
+  );
+  pool.addDestinationToken(crossToken);
+
+  const acc = sdk.Accounts.loadAccount(event.transaction.from);
+  acc.transferOut(
+    pool,
+    pool.getDestinationTokenRoute(crossToken)!,
+    event.params.to,
+    event.params.amount,
+    event.transaction.hash
+  );
+  pool.addRevenueNative(event.params.protocolFee, event.params.supplyFee);
+}
+```
+
+### Set Reward Emissions
+
+```typescript
+const rewardTokenAddress = "0x0.....";
+
+export function handlePoolRewardsUpdated(event: PoolRewardsUpdated): void {
+  const sdk = new SDK(conf, new Pricer(), new TokenInit(), event);
+
+  const pool = sdk.Pools.loadPool(event.params.poolAddress);
+  const rewardToken = sdk.Tokens.initToken(rewardTokenAddress);
+  const dailyEmissions = calculateDailyEmissions(
+    event.params.emissionsPerBlock
+  );
+
+  pool.setRewardEmissions(RewardTokenType.DEPOSIT, rewardToken, dailyEmissions);
+}
+```
+
+## A Note On Snapshots
+
+You don't need to worry about snapshots anymore. At all. But you might want to know how they work under the hood.
+Snapshots from each entity that has them are taken when the given entity is loaded. If the time since the last snapshot was taken is longer than the snapshot period, a snapshot will be taken with the data the entity had at the time of the load. Any changes made to an entity after a load will be part of the next snapshot.
```

Comment:
I made it this way for 2 reasons:
- It’s easier to implement (now that snapshots are immutable)
- Taking the snapshot on save would potentially lead to some events being more than 1h/1d apart being part of the same snapshot. Since snapshots are immutable, and because we take them when the period has elapsed, taking them on some entity.save() would include the changes of that entity together with all the previous ones even if a whole month elapsed.

Reply:
Makes sense!